ci/travis/build-docker-images.py

```diff
@@ -0,0 +1,197 @@
+    built_images = []
+    for gpu in ["-cpu", "-gpu"]:
+        if image_name == "base-deps":
+            gpu_arg = "BASE_IMAGE={}".format(
+                "nvidia/cuda:10.1-cudnn8-runtime-ubuntu18.04"
+                if gpu else "ubuntu:focal")
+        else:
+            gpu_arg = f"GPU={gpu}"
+
+        if "ray" in image_name:
+            wheel_arg = "--build-arg WHEEL_PATH=.whl/" + _get_wheel_name()
+        else:
+            wheel_arg = ""
+
+        tagged_name = f"rayproject/{image_name}:nightly:{gpu}"
+        build_command = (
+            f"docker build --no-cache --build-arg {gpu_arg} {wheel_arg} "
+            f"-t {tagged_name} {_get_root_dir()}/docker/{image_name}")
+
+        _subprocess_wrapper(build_command)
+        print("BUILT: ", tagged_name)
+        built_images.append(tagged_name)
+    return built_images
+
+
+def copy_wheels():
+    root_dir = _get_root_dir()
+    wheel = _get_wheel_name()
+    source = os.path.join(root_dir, ".whl", wheel)
+    shutil.copy(source, os.path.join(root_dir, "docker/ray/.whl/"))
+    shutil.copy(source, os.path.join(root_dir, "docker/ray-deps/.whl/"))
+
+
+def build_or_pull_base_images(is_docker_affected: bool) -> List[str]:
+    """Returns images to tag and build"""
+    _subprocess_wrapper("docker pull rayproject/base-deps:nightly")
+    age = _subprocess_wrapper(
+        "docker inspect -f '{{ .Created }}' rayproject/base-deps:nightly")
+    short_date = datetime.datetime.strptime(age.split("T")[0], "%Y-%m-%d")
+    is_stale = (
+        datetime.datetime.now() - short_date) > datetime.timedelta(days=14)
+
+    if is_stale or is_docker_affected or _release_build():
+        for image in ["base-deps", "ray-deps"]:
+            _build_helper(image)
+        return True
+    else:
+        print("Just pulling images!")
+        _subprocess_wrapper("docker pull rayproject/base-deps:nightly-cpu")
+        _subprocess_wrapper("docker pull rayproject/ray-deps:nightly-gpu")
+        _subprocess_wrapper("docker pull rayproject/ray-deps:nightly-cpu")
+        return False
+
+
+def build_ray() -> None:
+    return _build_helper("ray")
+
+
+def build_ray_ml() -> None:
+    root_dir = _get_root_dir()
+    requirement_files = _subprocess_wrapper(
+        f"basename {_get_root_dir()}/python/requirements*.txt").split()
+    for fl in requirement_files:
+        shutil.copy(
+            os.path.join(root_dir, "python/", fl),
+            os.path.join(root_dir, "docker/ray-ml/"))
+    ray_ml_images = _build_helper("ray-ml")
+    for img in ray_ml_images:
+        _subprocess_wrapper(
+            f"docker tag {img} {img.replace('ray-ml', 'autoscaler')}")
+
+
+# For non-release builds, push "nightly" & "sha"
+# For release builds, push "nightly" & "latest" & "x.x.x"
+def push_and_tag_images(push_base_images: bool) -> None:
+    if _merge_build():
+        _subprocess_wrapper(
+            f"echo \"$DOCKER_PASSWORD\" | " +
+            "docker login -u {DOCKER_USERNAME} --password-stdin")
```

Comment:
not sure, but don't we want to use this?
```suggestion
            "echo \"$DOCKER_PASSWORD\" | " +
            f"docker login -u {DOCKER_USERNAME} --password-stdin")
```

Reply:
Thanks!! Yes I do!